pandas/core/groupby/grouper.py

```diff
@@ -517,7 +517,7 @@ def _get_grouper(
         if key.key is None:
             return grouper, [], obj
         else:
-            return grouper, {key.key}, obj
+            return grouper, [key.key], obj
```

Comment:
So is there no impact changing this from a set to a list? If not, does it even need to be a container?

Reply:
This just makes the types consistent for all returns. A set or a list is not important when there's only 1 item in the container.

Reply:
BTW It sometimes has lists with several items, so needs to be a list/container.

Reply:
Hmm not sure I follow. Maybe better to ask what the error was that mypy was giving? I assume this gets mutated outside of the function right?

Reply:
Oh, it wasn't a mypy error, I think the function should have a clear return type, so ``List`` rather than ``Contaioner``.

(More general): I think function may have broad parameter types, but should strive to have clear return types, so there's less ambiguity over what you're looking at. Don't you agree?

Reply:
OK. Yea I'm not necessarily saying the old container type was correct and this is wrong, I guess was just questioning why this even needs to be held in a container if it's simply one value being returned.

Can be refactored or reviewed in a follow up if that's a big effort; not a blocker as is

Reply:
Yeah, I'm haven't checked if a single value could be returned, instead of the list.

My motivation for doing this is mostly wanting to pin down what I'm looking at in ``groupBy``, ``Grouper`` etc. I think "real" clean-ups must come later. It's quite complex to me now, and I'm hoping these typeups will ease understanding.

Reply:
Sounds good. Yea this part of code is tricky and could use some refactor, so this is all moving in the right direction